app/org/sagebionetworks/bridge/services/SchedulePlanServiceImpl.java

```diff
@@ -59,23 +55,29 @@ public SchedulePlan getSchedulePlan(StudyIdentifier studyIdentifier, String guid
     }
 
     @Override
-    public SchedulePlan createSchedulePlan(SchedulePlan plan) {
+    public SchedulePlan createSchedulePlan(Study study, SchedulePlan plan) {
         checkNotNull(plan);
-        
+
+        // Plan must always be in user's study
+        plan.setStudyKey(study.getIdentifier());
+
```

Comment:
I have been silently setting healthCode and studyKey to the correct values as these shouldn't be exposed in the API, and even if someone tries to force it with a JSON payload, we should just undo any attempt to write to a different study or a different account. This also makes it easier to copy something between projects later, if we do that.